### DISCUSSION TITLE
Unify cart validity checking functions and improve it

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#33423] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

While making https://github.com/PrestaShop/PrestaShop/pull/26048/ PR, I worked with some unoptimized code:
```
$allProductsInStock = $cart->isAllProductsInStock();
$allProductsExist = $cart->checkAllProductsAreStillAvailableInThisState();
$allProductsHaveMinimalQuantity = $cart->checkAllProductsHaveMinimalQuantities();
```

There is also:
`$this->context->cart->checkQuantities(true);`

### Describe the solution you'd like

- These functions should be explored, so we know what they do and why they are separate
- They should be unified to a single function not to load cart product list 3 times
- They should also check for cart rule and catalog rule validity.

_______________

Ping @tswfi, I can't add TODOs to the code 😄

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon.

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
